test/serializers/associations_test.rb

```diff
@@ -290,6 +289,82 @@ def test_illegal_conditional_associations
           assert_match(/:if should be a Symbol, String or Proc/, exception.message)
         end
       end
+
+      class InheritedSerializerTest < ActiveSupport::TestCase
+        InheritedPostSerializer = Class.new(PostSerializer) do
+          belongs_to :author, polymorphic: true
+          has_many :comments, key: :reviews
+        end
+
+        InheritedAuthorSerializer = Class.new(AuthorSerializer) do
+          has_many :posts, polymorphic: true
+          has_many :roles, polymorphic: true
+          has_one :bio, key: :biography
+        end
+
+        def setup
+          @author = Author.new(name: 'Steve K.')
+          @post = Post.new(title: 'New Post', body: 'Body')
+          @post_serializer = PostSerializer.new(@post, custom_options: true)
+          @author_serializer = AuthorSerializer.new(@author)
+          @inherited_post_serializer = InheritedPostSerializer.new(@post, custom_options: true)
+          @inherited_author_serializer = InheritedAuthorSerializer.new(@author)
+        end
+
+        def map_with_key(associations)
+          associations.map { |a| a.options.fetch(:key, a.name) }
+        end
+
+        def test_redefined_has_many_and_has_one
+          author_associations = @author_serializer.associations.to_a
+          inherited_author_associations = @inherited_author_serializer.associations.to_a
+          assert_equal(
+            @author_serializer.associations.map(&:name).sort,
+            [:posts, :roles, :bio].sort
+          )
+          assert_equal(
+            @inherited_author_serializer.associations.map(&:name).sort,
+            [:posts, :roles, :bio, :bio].sort
+          )
+          assert_equal(
+            map_with_key(@inherited_author_serializer.associations).sort,
+            [:posts, :roles, :bio, :biography].sort
```

Comment:
These testes look MUCH better. :-)

Just a couple more changes, and I think I'll be ready to merge.
1. I find myself having to look at your test setup, and back at the different assertions to figure out what is being tested. Do you think you could break up some of these assertions into different tests?

Something like:

```ruby
test 'a serializer inheriting from the author serializer can have an additional bio field named biography' do
  expected = [:bio, :biography, :posts, :roles]
  result = map_with_key(@inherited_author_serializer.associations).sort
end
```

that `test 'name of test' do` syntax is also what we're trying to get people to do going forward, but yeah.


So, if you could change the tests to be like that, that'd be great! I know it's more computing of the same thing for each test, but readability for others looking at the project is what the goal is here. :-)


the second thing, is just to add a changelog entry :-)